### PR TITLE
Unit tests for the backend API

### PR DESCRIPTION
**Pull request by marmoure**
This PR adds tests for the GitHub router and the GitHub controller.

# how to test this PR
* checkout the branch
* install new dependencies `yarn instal`
* run the tests `yarn run test:backend`
* check the backend test on CI

### CODE REVIEW

File: packages/prosemirror-lwdita-backend/src/api/controller/github.controller.ts [hunk 1]
```diff
@@ -33,6 +33,9 @@ export const authenticateUserWithOctokit = async (req: Request, res: Response) =
   // dynamically import the module to avoid EMS and CJS incompatibility
   const { authenticateWithOAuth } = await import('../modules/octokit.module.mjs');
 
+  console.log("code", code);
```
Comment (plutonik-a) [inline packages/prosemirror-lwdita-backend/src/api/controller/github.controller.ts:36]: Is this log still needed?

File: packages/prosemirror-lwdita-backend/src/tests/github.controller.spec.ts [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+import { authenticateUserWithOctokit, commitChangesAndCreatePR, getUserInformation } from '../api/controller/github.controller';
+
+describe('authenticateUserWithOctokit', () => {
+  let req: Partial<Request>;
+  let res: Partial<Response>;
+  let jsonStub: sinon.SinonStub;
+  let statusStub: sinon.SinonStub;
+
+  beforeEach(async () => {
+    // Mocking req and res objects
```
Comment (plutonik-a) [inline packages/prosemirror-lwdita-backend/src/tests/github.controller.spec.ts:30]: Please remove comments that don't add any value from the test.

File: packages/prosemirror-lwdita-backend/src/tests/github.controller.spec.ts [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+});
+
+describe('getUserInformation', () => {
+  let req: Partial<Request>;
+  let res: Partial<Response>;
+  let jsonStub: sinon.SinonStub;
+  let statusStub: sinon.SinonStub;
+
+  beforeEach(async () => {
+    // Mocking req and res objects
```
Comment (plutonik-a) [inline packages/prosemirror-lwdita-backend/src/tests/github.controller.spec.ts:86]: Same as above

File: packages/prosemirror-lwdita-backend/src/tests/github.controller.spec.ts [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+  });
+
+  it.skip('should return user information when token is provided', async () => {
+    // Arrange: Set req.headers.authorization
+    if (!req.headers) {
+      req.headers = {};
+    }
+    req.headers.authorization = 'Bearer test_token';
+
+    // Act: Call the controller
```
Comment (plutonik-a) [inline packages/prosemirror-lwdita-backend/src/tests/github.controller.spec.ts:121]: Same as above

File: packages/prosemirror-lwdita-backend/src/tests/github.controller.spec.ts [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+    await getUserInformation(req as Request, res as Response);
+    //FIXME(YB): Unable to mock getUserInfo function
+
+    // Assert: Check response status and message
```
Comment (plutonik-a) [inline packages/prosemirror-lwdita-backend/src/tests/github.controller.spec.ts:125]: Same as above

File: packages/prosemirror-lwdita-backend/src/tests/github.controller.spec.ts [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+  });
+
+  afterEach(() => {
+    sinon.restore(); // Restore original functionality
+  });
+
+  it('should return 403 if authentication header is missing', async () => {
+    // Arrange: Leave req.headers.authorization undefined
+
+    // Act: Call the controller
```
Comment (plutonik-a) [inline packages/prosemirror-lwdita-backend/src/tests/github.controller.spec.ts:106]: same as above

File: packages/prosemirror-lwdita-backend/src/tests/github.controller.spec.ts [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+
+describe('commitChangesAndCreatePR', () => {
+  let req: Partial<Request>;
+  let res: Partial<Response>;
+  let jsonStub: sinon.SinonStub;
+  let statusStub: sinon.SinonStub;
+
+
+  beforeEach(async () => {
+    // Mocking req and res objects
```
Comment (plutonik-a) [inline packages/prosemirror-lwdita-backend/src/tests/github.controller.spec.ts:140]: same as above

**Review by plutonik-a — CHANGES_REQUESTED** [review]
Looking good to me, but please remove all unnecessary comments from the test.

File: packages/prosemirror-lwdita-backend/src/tests/github.controller.spec.ts (hunk 4 above)
Comment (marmoure) [reply packages/prosemirror-lwdita-backend/src/tests/github.controller.spec.ts:121]: So these comments: `Arrange` ,`ACT`, `Assert` are really connected to the stubbing methodology and I strongly recommend we should keep them in the tests.

File: packages/prosemirror-lwdita-backend/src/tests/github.controller.spec.ts (hunk 5 above)
Comment (marmoure) [reply packages/prosemirror-lwdita-backend/src/tests/github.controller.spec.ts:125]: these comments: Arrange ,ACT, Assert are really connected to the stubbing methodology and I strongly recommend we should keep them in the tests.

File: packages/prosemirror-lwdita-backend/src/tests/github.controller.spec.ts (hunk 6 above)
Comment (marmoure) [reply packages/prosemirror-lwdita-backend/src/tests/github.controller.spec.ts:106]: these comments: Arrange ,ACT, Assert are really connected to the stubbing methodology and I strongly recommend we should keep them in the tests.

File: packages/prosemirror-lwdita-backend/src/tests/github.controller.spec.ts (hunk 7 above)
Comment (marmoure) [reply packages/prosemirror-lwdita-backend/src/tests/github.controller.spec.ts:140]: these comments: Arrange ,ACT, Assert are really connected to the stubbing methodology and I strongly recommend we should keep them in the tests.

File: packages/prosemirror-lwdita-backend/src/tests/github.controller.spec.ts (hunk 3 above)
Comment (marmoure) [reply packages/prosemirror-lwdita-backend/src/tests/github.controller.spec.ts:86]: these comments: Arrange ,ACT, Assert are really connected to the stubbing methodology and I strongly recommend we should keep them in the tests.

File: packages/prosemirror-lwdita-backend/src/tests/github.controller.spec.ts (hunk 2 above)
Comment (marmoure) [reply packages/prosemirror-lwdita-backend/src/tests/github.controller.spec.ts:30]: these comments: Arrange ,ACT, Assert are really connected to the stubbing methodology and I strongly recommend we should keep them in the tests.

File: packages/prosemirror-lwdita-backend/src/api/controller/github.controller.ts (hunk 1 above)
Comment (marmoure) [reply packages/prosemirror-lwdita-backend/src/api/controller/github.controller.ts]: No, Fixed in 625fd3a3319e6d0da09f769cda2e88480cfa4769

**Pull request merged**